internal/cli/cmd/verify/stage.go

```diff
@@ -0,0 +1,134 @@
+package verify
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"connectrpc.com/connect"
+	"github.com/spf13/cobra"
+
+	"github.com/akuity/kargo/internal/cli/client"
+	"github.com/akuity/kargo/internal/cli/config"
+	"github.com/akuity/kargo/internal/cli/option"
+	v1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+type verifyStageOptions struct {
+	Config        config.CLIConfig
+	ClientOptions client.Options
+
+	Project string
+	Name    string
+	Abort   bool
+}
+
+func newVerifyStageCommand(cfg config.CLIConfig) *cobra.Command {
+	cmdOpts := &verifyStageOptions{
+		Config: cfg,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "stage [--project=project] (NAME) [--abort]",
+		Short: "(Re)run or abort the verification of the stage's current freight",
+		Args:  option.ExactArgs(1),
+		Example: `
+# Rerun the verification of the stage's current freight
+kargo verify stage --project=my-project my-stage
+
+# Rerun the verification of a stage in the default project
+kargo config set-project my-project
+kargo verify stage my-stage
+
+# Abort the verification of a stage's current freight
+kargo verify stage --project=my-project my-stage --abort
```

Comment:
I'm a little iffy on `--abort` as a flag on `kargo verify stage`.

We've gotten a lot better about consistently adhering to `kargo verb nound` and this feels like a departure from that.

I get why you did it this way. There's currently nothing else "abort" applies to, which would make it feel like some awkward one-off sub-command...

I feel we will soon have _other_ things that can be aborted as well. I know @jessesuen has mentioned aborting promotions a time or two before...

We might want to plan ahead for that and consider making `abort` its own subcommand after all.

Reply:
What would the full abort version be called in this case? As with `kargo abort verification --stage <stage>`, I feel it would be counterintuitive from `kargo verify stage`.

Reply:
Ya... now that you say that... that seems clunky because `verification` isn't a resource -- nor do I think it needs to be.

Obviously this was non-blocking anyway.